src/components/show_or_hide.tsx

```diff
@@ -7,38 +7,50 @@ interface ShowOrHideProps {
     visible: boolean
     path?: string
     value?: string
+    parent?: any
 }
 
 /**
- * `ShowOrHide` is a React component that conditionally renders its children based on
- * the `visible` prop and optionally checks a value from the `StoreContext` against
- * a specified path.
+ * `ShowOrHide` is a React component that conditionally renders its children based on the `visible` prop
+ * and optionally checks a value from the `StoreContext` or a parent document against a specified path.
  *
- * @param {Object} props - The component's props.
- * @param {React.ReactNode} props.children - The children elements to be conditionally rendered.
- * @param {boolean} props.visible - A boolean that directly determines if the children should be shown.
- * @param {string} [props.path] - An optional path to check within the context's data.
- * @param {any} [props.value] - An optional value to compare against the value at the specified path.
+ * If the `visible` prop is `true`, the children will always be rendered. If `visible` is `false`, the component
+ * will check the specified `path` within the context data or parent data and compare it to the `value` prop.
+ * If the value at the `path` matches the `value` prop, the children will be rendered. Otherwise, the children are hidden.
  *
- * @returns {React.ReactElement} The rendered component.
+ * @param {React.ReactNode} children - The child components to be conditionally rendered.
+ * @param {boolean} visible - A boolean that directly determines whether the children should be rendered.
+ * @param {string} [path] - An optional path to check within the context's data.
+ * @param {any} [value] - An optional value to compare against the value at the specified `path` in the context data.
+ * @param {any} [parent] - An optional parent document, used to override the context data if provided.
+ *
+ * @returns {React.ReactElement} - The rendered component, either the children or null, based on the visibility logic.
+ *
+ * @example
+ * <ShowOrHide visible={true}>
+ *   <p>This content is always visible</p>
+ * </ShowOrHide>
+ *
+ * <ShowOrHide path="role" value="installer">
+ *   <p>This content is visible only if the user's role is 'installer'</p>
+ * </ShowOrHide>
  */
 const ShowOrHide: FC<ShowOrHideProps> = ({
```

Comment:
Can the `ShowOrHide` component have a prop for whether the value determines if the children should be shown or hidden? This is different to the behavior of the `visible` prop, which forces the visibility of the children.

For example, a template may want to hide a panel if a given value is selected.

Reply:
@markborkum The component is enhanced to check this option.